.github/workflows/ci.yaml

```diff
@@ -91,3 +91,50 @@ jobs:
           retention-days: 1
           name: image-tar-frrk8s
           path: /tmp/frrk8s.tar
+  e2e:
+    runs-on: ubuntu-22.04
+    needs:
+      - unit-tests
+      - build-test-images
+      - commitlint
+    strategy:
+      fail-fast: false
+      matrix:
+        ip-family: [ipv4, ipv6, dual]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install kernel modules
+        run: |
+          sudo apt-get update
+          sudo apt-get install linux-modules-extra-$(uname -r)
+
+      - name: Download frr8ks images
+        uses: actions/download-artifact@v3
+        with:
+          path: image
+
+      - name: Load image
+        working-directory: image
+        run: |
+          docker load -i image-tar-frrk8s/frrk8s.tar
+
+      - name: E2E
+        run: |
+          IP_FAMILY="${{ matrix.ip-family }}"
+          IP_FAMILY=$IP_FAMILY make kind-cluster
+          SKIP="none"
+          if [ "${{ matrix.ip-family }}" == "ipv4" ]; then SKIP="$SKIP|IPV6|DUALSTACK"; fi
+          if [ "${{ matrix.ip-family }}" == "dual" ]; then SKIP="$SKIP|IPV6"; fi
```

Comment:
Because a dual stack cluster still keeps the default ip as ipv4. So it behaves as a regular ipv4 cluster PLUS ipv6 (unless you configure it to have ipv6 as the primary ip).
Because of this, all ipv4 tests still apply (but not viceversa)